Refactor hydro series

payetvin opened the pull request:
#1669

payetvin committed:
[DEV] Removed hydro/series.hxx

payetvin committed:
[DEV] Use a ref for TSNumbers

payetvin committed:
[DEV] Use a ref for TSNumbers (#1706)

payetvin committed:
[DEV] Use TimeSeries for ror

payetvin committed:
Merge branch 'feature/series-hydro' of https://github.com/AntaresSimulatorTeam/Antares_Simulator into feature/series-hydro

sylvlecl commented on src/solver/hydro/management/management.cpp:
```diff
@@ -380,7 +380,7 @@ void HydroManagement::prepareNetDemand(uint numSpace, uint year)
         auto& rormatrix = area.hydro.series->ror;
         auto tsIndex = area.hydro.series->getIndex(year);
-        auto& ror = rormatrix[tsIndex < rormatrix.width ? tsIndex : 0];
+        const auto* ror = rormatrix.getColumn(tsIndex < rormatrix.timeSeries.width ? tsIndex : 0);
```
I think the `getColumn` already handles the indirection : we don't need any more to first get the index and use it, we should use the year instead here.
```cpp
const auto* ror = rormatrix.getColumn(year);
```

payetvin committed:
[DEV] Modify operator[] behavior

payetvin committed:
[DEV] storage

payetvin committed:
[CI] Upgrade python to 3.12

payetvin committed:
[DEV] mingen

payetvin committed:
[DEV] replace tsindex with year in hydro/management

payetvin committed:
[DEV] Remove getIndex from hydro series

payetvin committed:
[DEV] code smells

guilpier-code committed:
Make hydro ts count a private member (#1712)

* Refactor max power TS checks : internalize count of generation time series (ror, storage, mingen)

We should not be able to change a TS count without resizing the TS.
This was allowed when because this count was a public data member.
So we make it a private data member, and change it only when resizing the associated data.

* Make hydro time series cont a private member
